websocket/controller.cc

```diff
@@ -35,16 +27,34 @@ future<> controller::start_server()
     }
     _active = true;
     co_await _server.start();
-    co_await _server.invoke_on_all([] (experimental::websocket::server& ws) {
-        ws.register_handler("echo", [] (input_stream<char>& in, output_stream<char>& out) -> future<> {
-            while (true) {
-                temporary_buffer<char> f = co_await in.read();
-                if (f.empty()) {
-                    co_return;
+    co_await _server.invoke_on_all([this] (experimental::websocket::server& ws) {
+        ws.register_handler("cql", [this] (input_stream<char>& in, output_stream<char>& out) -> future<> {
+            auto unix_server_socket = seastar::listen(socket_address{unix_domain_addr{""}});
+            auto unix_addr = unix_server_socket.local_address();
+            auto accept_fut = unix_server_socket.accept();
+            auto sock = co_await connect(unix_addr);
+            auto accept_result = co_await std::move(accept_fut);
+            auto connection = _cql_server.local().inject_connection(unix_addr,
```

Comment:
The `_cql_server` thing can stop on-the-fly because transport server is stoppable via the API call. I have concern that these two play nicely with each other

Reply:
Indeed, that's my main concern about forwarding requests via an internal unix domain socket. I guess in order to get the shutdown/drain order straight, this websocket server needs to be able to stop before CQL, e.g. with an abort_source, and always assume that the server might have stopped already.